mailpoet/assets/js/src/settings/store/create_reducer.ts

```diff
@@ -16,17 +16,23 @@ export function createReducer(defaultValue: State) {
     action: Action,
   ): State => {
     switch (action.type) {
-      case 'SET_SETTING':
-        return setWith(
+      case 'SET_SETTING': {
+        const newState = setWith(
           clone(state),
           ['data', ...action.path],
           action.value,
           clone,
         );
+
+        newState.hasUnsavedChanges = true;
+
+        return newState;
+      }
       case 'SET_SETTINGS':
         return {
           ...state,
           data: normalizeSettings(action.value as Record<string, unknown>),
+          hasUnsavedChanges: false,
```

Comment:
I think understand why is the reset of `hasUnsavedChanges` done here, but I find it a bit confusing because without searching for usages of 'SET_SETTINGS' action just from naming I would assume that this action should set the `hasUnsavedChanges` to the opposite.
Maybe we could rename the action to `SET_SETTINGS_FROM_SERVER` or `SET_SAVED_SETTINGS` so that it is more obvious. Or another option could be resetting the flag on `SAVE_DONE` action. That would require yielding the action also [in `verifyMssKey` action](https://github.com/mailpoet/mailpoet/blob/d116ff360b8d73f0ae09429ae0c365ed30a54def/mailpoet/assets/js/src/settings/store/actions/mss_and_premium.ts#L14).
What do you think?

Reply:
I agree that the name `SET_SETTINGS` can be misleading and it confused me when I first read the code. That being said, I opted to yield `SAVE_DONE` in `verifyPremiumKey()` as I wasn't sure how complex it would be to rename `SET_SETTINGS` and also just renaming it is not enough. When the user clicks on the `Verify` button, after `SET_SETTINGS` runs, `SET_SETTING` also runs changing state.save.hasUnsavedChanges to true.